rs-matter/src/transport/network/btp/gatt.rs

```diff
@@ -0,0 +1,252 @@
+    }
+
+    /// Return an iterator over the binary representation of the AD1 advertising data (Flags).
+    /// Useful with GATT stacks that require the advertising data to be reported as separate AD records
+    pub fn flags_iter(&self) -> impl Iterator<Item = u8> + '_ {
+        empty()
+            .chain(once(self.flags_payload_iter().count() as u8 + 1)) // 1-byte type
+            .chain(once(self.flags_adv_type()))
+            .chain(self.flags_payload_iter())
+    }
+
+    /// The AD1 advertising data type (Flags).
+    pub const fn flags_adv_type(&self) -> u8 {
+        0x01
+    }
+
+    /// Return an iterator over the binary representation of the AD1 advertising data _payload_.
+    /// Useful with GATT stacks that require the advertising data to be reported as separate AD records
+    pub fn flags_payload_iter(&self) -> impl Iterator<Item = u8> + '_ {
+        once(0x06)
+    }
+
+    /// Return an iterator over the binary representation of the AD2 advertising data (UUID16+Service Data).
+    pub fn service_iter(&self) -> impl Iterator<Item = u8> + '_ {
+        empty()
+            .chain(once(self.service_payload_iter().count() as u8 + 3)) // + 1-byte type and 2-bytes Matter UUID16 Service
+            .chain(once(self.service_adv_type()))
+            .chain(MATTER_BLE_SERVICE_UUID16.to_le_bytes())
+            .chain(self.service_payload_iter())
+    }
+
+    /// The AD2 advertising data type (UUID16+Service Data).
+    pub const fn service_adv_type(&self) -> u8 {
+        0x16
+    }
+
+    /// Return an iterator over the binary representation of the AD2 advertising data _payload_.
+    /// Useful with GATT stacks that require the advertising data to be reported as separate AD records
+    pub fn service_payload_iter(&self) -> impl Iterator<Item = u8> + '_ {
+        [
+            0, // Always 0 = "Commissionable"
+            self.discriminator.to_le_bytes()[0],
+            self.discriminator.to_le_bytes()[1],
+            self.vid.to_le_bytes()[0],
+            self.vid.to_le_bytes()[1],
+            self.pid.to_le_bytes()[0],
+            self.pid.to_le_bytes()[1],
+            0, // No additional data
+        ]
+        .into_iter()
+    }
+}
+
+/// A minimal GATT peripheral event.
+/// This enum is used to abstract the platform-specific GATT peripheral events.
+///
+/// The abstraction is "minimal" in the sense that it is good enough for the purposes of
+/// the Matter BTP protocol, but is otherwise not really having the ambition to model all
+/// possible events of a generic GATT peripheral, which would result in a much larger API surface.
+#[derive(Debug, Clone)]
+pub enum GattPeripheralEvent<'a> {
+    /// A GATT central has subscribed for notifications from characteristic `C2`.
+    /// In other words, the GATT central is now ready to receive BTP packets.
+    ///
+    /// See the Matter Core spec w.r.t. details on characteristic `C2`.
+    NotifySubscribed(BtAddr),
```

Comment:
I'm not so sure we should be doing that! :)

See above my previous comment. How much _more specific_ should we be here given that we explicitly say:
```
/// The abstraction is "minimal" in the sense that it is good enough for the purposes of
/// the Matter BTP protocol, but is otherwise not really having the ambition to model all
/// possible events of a generic GATT peripheral, which would result in a much larger API surface.
```

`NotifySubscribed` here means the peer has done a certain set of low-level operations which ended up with the _higher level notion of BTP "subscription"_. As in doing a cccd write on characteristics C2 of the GATT peripheral and the peripheral remembering this.

Ditto for `NotifyUnsubscribed` and `Write`. The only way a peer can send data to us is via the `C1` characteristics, period. Why should I encode that in the name, given that the comments are laboriously clear on that already? 

(And this write btw should be a confirmed write so the GATT peripheral is supposed to confirm it so the higher level notion of "write" here is also a pair (or more) of potentially smaller GATT lower level ops.)

And then again renaming `Write` to `C1Write` can be (mis)interpreted as if the peer can write to C2 as well, or can subscribe via `C1` as well, which is just not the case - as per my earlier comment.